src/plugins/analysis/plugin.py

```diff
@@ -0,0 +1,393 @@
+        pass
+
+
+class PluginV1(metaclass=abc.ABCMeta):
+    """An abstract class that all analysis plugins must inherit from.
+
+    This class MAY NOT depend on FACT_core code.
+    """
+
+    @dataclass
+    class MetaData:
+        """A class containing all metadata that describes the plugin"""
+
+        #: Name of the plugin
+        name: str
+        #: The plugins description.
+        #: MUST be a single sentence.
+        description: str
+        #: The version of the plugin.
+        #: MUST adhere to semantic versioning.
+        version: str
+        #: A list of all plugins that this plugin depends on
+        dependencies: list[str]
+        #: List of mimetypes that should not be processed
+        mime_blacklist: list
+        #: List of mimetypes that should be processed
+        mime_whitelist: list
+        #: Pydantic dataclass of the object returned by :py:func:`analyse`.
+        #: Currently unused, will be used in the future.
+        # pylint: disable=invalid-name
+        Schema: typing.Type = None
+
+        # pylint: disable=no-self-argument
+        @validator('version')
+        def _version_validator(cls, value):
+            is_semver = re.fullmatch(
+                # The regex from semver.org
+                r'^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$',
+                value,
+            )
+            if not is_semver:
+                raise ValueError('Version must use semantic versioning')
+
+    def __init__(self, metadata: MetaData):
+        self.metadata: PluginV1.MetaData = metadata
+
+    # The type MetaData.Schema
+    Schema = typing.TypeVar('T')
+
+    @abc.abstractmethod
+    def summarize(self, result: Schema) -> list[str]:
+        """Returns a list of  strings that summarizes the analysis.
+        Will only be called if analyze did not return None.
+
+        A summary most often is either just a single sentence or a list of words.
+        Some examples of summarys are:
+
+            * From the software_components plugin: ``["BusyBox 1.29.3", "Linux Kernel 4.9.250", "SQLite 3.8.11.1"]``
+            * ``["Nothing found"]``
+            * ``["SSLCertificate"]``
+
+        :param result: The analysis as returned by :py:func:`analyze`
+        """
+
+    @abc.abstractmethod
+    def analyze(self, file_handle: io.FileIO, virtual_file_path: dict, analyses: dict[str, dict]) -> Schema:
+        """Analyze a file.
+
+        :param file_handle: :py:class:`io.FileIO` instance of the file to be analyzed
+        :param virtual_file_path: The virtual file paths, see :py:class:`~objects.file.FileObject`
+        :param analyses: A dictionary of dependent analysis
+        """
+
+    @typing.final
+    def get_analysis(self, file_handle: io.FileIO, virtual_file_path: dict, analyses: dict[str, dict]) -> dict:
```

Comment:
I still don't like the similar names of `analyze` and `get_analysis`. Any ideas?